plugins/unicodelib/unicodelib.cpp

```diff
@@ -719,6 +720,34 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+void translate(UnicodeString & source, UChar const * sear, unsigned searLen, UChar const * repl, unsigned replLen)
+{
+    UnicodeString search(sear, searLen);
+    UnicodeString replace(repl, replLen);
+    if (search.countChar32() != replace.countChar32() || source.isEmpty() || search.isEmpty() || replace.isEmpty())
+    {
+        return;
+    }
+    StringCharacterIterator it(source);
+    source.remove();
+    int32_t idx = it.setToStart();
+    while (idx != it.endIndex())
+    {
+        int32_t x = search.lastIndexOf(it.current32());
```

Comment:
This is what the code would look like if you are using next32PostInc():

    it.setToStart();
    for(;;)
    {
        int32_t next = it.next32PostInc();
        if (next == CharacterIterator::DONE)
            break;
        int32_t x = search.lastIndexOf(next);
        if (x == -1)
        {
            toProcess.append(next);
        }
        else
        {
            x = replace.moveIndex32(0, x);
            toProcess.append(replace.char32At(x));
        }
    }